pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,200 @@ func TestAccessTokenFile(t *testing.T) {
-		Convey("Returns no error if Data input is not nil", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
-			err := tokenInFile.Write(dataActual)
 
-			So(err, ShouldEqual, nil)
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method, writes the access token to a file in the path defined", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite1")
+					So(err, ShouldEqual, nil)
+				})
+			})
 		})
 
-		Convey("Returns error if Data input is nil", func() {
-			err := tokenInFile.Write(nil)
+		Convey("Given an access token without data", func() {
 
-			So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+			Convey("When running the Write method and attempts to write the access token", func() {
+				err := tokenInFile.Write(nil)
+
+				Convey("Raises an error that the access token data is empty", func() {
+					So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+				})
+			})
 		})
 
-		Convey("Returns no error if file already exists", func() {
+		Convey("Given access token with data and a file location defined", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite2"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
 
-			// Second time
-			dataActual = []byte{'t', 'e', 's', 't', '2'}
-			err = tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+			})
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method a second time", func() {
+				dataActual = []byte{'t', 'e', 's', 't', '2'}
+				err := tokenInFile.Write(dataActual)
+				So(err, ShouldEqual, nil)
 
-			// Test we are reading the new Data
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
+				// Check if file exits
+				_, err = os.Stat("/tmp/accessTokenFileWrite2")
+				So(err, ShouldEqual, nil)
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+				Convey("Raises no error and the data should be written to the file as expected", func() {
```

Comment:
please split to 2 conveys

Reply:
I don't think these should be broken up into 2 because the test isn't focused on reading, it's focused on writing... I think instead of checking if Read was done correct it will be:
```
Convey("Writes the data the file as expected", func() {
 	dataExpected, _ := tokenInFile.Read()
	eq := reflect.DeepEqual(dataActual, dataExpected)
	So(eq, ShouldEqual, true)
})
```

and just get rid of the check of `err` because if Read wasn't done correctly, it will fail there

Reply:
good point